src/Microsoft.AspNet.Mvc.Core/Filters/AsyncTimeoutAttribute.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc.Core;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Represents an attribute that is used to set the timeout value, in milliseconds, 
+    /// which when elapsed will cause the current request to be aborted.
+    /// </summary>
+    [AttributeUsage(
+        AttributeTargets.Class | AttributeTargets.Method,
+        AllowMultiple = false,
+        Inherited = true)]
+    public class AsyncTimeoutAttribute : Attribute, IAsyncResourceFilter
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="AsyncTimeoutAttribute"/>.
+        /// </summary>
+        /// <param name="duration">The duration in milliseconds.</param>
+        public AsyncTimeoutAttribute(int duration)
```

Comment:
Some people would say this parameter should be named `durationInMilliseconds` (and same for the property). Thoughts?